radix-engine/src/kernel/interpreters/scrypto_interpreter.rs

```diff
@@ -274,6 +253,17 @@ impl Executor for ScryptoExecutor {
             + ClientNativeInvokeApi<RuntimeError>,
         W: WasmEngine,
     {
+        // Make dependent resources/components visible
+        {
+            let handle = api.lock_substate(
+                RENodeId::Global(GlobalAddress::Package(self.package_address)),
+                NodeModuleId::SELF,
+                SubstateOffset::Package(PackageOffset::Info),
+                LockFlags::read_only(),
+            )?;
+            api.drop_lock(handle)?;
```

Comment:
Won't we lose the visiblity after `drop_lock`?

Reply:
Global (ie immovable) nodes don't lose visibility on drop locks.

Reply:
This is interesting, although we may need more thinking about "global" references.